osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs

```diff
@@ -123,6 +124,10 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                                 return SkinUtils.As<TValue>(new BindableFloat(legacy_circle_radius));
 
                             break;
+
+                        case OsuSkinConfiguration.ExpandNumberPiece:
+                            decimal legacyVersion = source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value ?? 1.0m;
```

Comment:
why 1.0m? where does this even come from? why not just 

`bool expand = source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value >= 2.0;`

Reply:
`value` could be null (if the provided skin itself provides skin.ini without version)? 

I thought you meant comparing null with non-null?

Reply:
you have changed the null handling logic. see my previous statement. the statement i gave you did not change this. confirm this was intentional

Reply:
previous code:
```csharp
decimal legacyVersion = source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value ?? 1.0m;
                            return SkinUtils.As<TValue>(new BindableBool(legacyVersion < 2.0m));
```
if Version is null or less than 2.0, expand

proposed code:
```csharp
return !(source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value >= 2.0);
```
if Version is null or less than 2.0, expand

your code:
```csharp
                            bool expand = source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value < 2.0m;
                            return SkinUtils.As<TValue>(new BindableBool(expand));
```
if Version is less than 2.0, expand

Reply:
I wrote it carelessly, after reading the docs regarding nullable comparsions, I understand now why you proposed the code to check if not greater or equal to 2.0, will fix it, sorry.